com.unity.visualeffectgraph/Editor/Core/VFXLibrary.cs

```diff
@@ -489,21 +482,66 @@ private static void LoadSRPBindersIfNeeded()
             }
         }
 
+        private static bool unsupportedSRPWarningIssued = false;
+
+        private static void LogUnsupportedSRP(VFXSRPBinder binder, bool forceLog)
+        {
+            if (binder == null && (forceLog || !unsupportedSRPWarningIssued))
+            {
+                Debug.LogWarning("The Visual Effect Graph is supported in the High Definition Render Pipeline (HDRP) and the Universal Render Pipeline (URP). Please assign your chosen Render Pipeline Asset in the Graphics Settings to use it.");
+                unsupportedSRPWarningIssued = true;
+            }
+        }
+
+        public static void LogUnsupportedSRP(bool forceLog = true)
+        {
+            bool logIssued = unsupportedSRPWarningIssued;
+            var binder = currentSRPBinder;
+
+            if (logIssued || !unsupportedSRPWarningIssued) // Don't reissue warning if inner currentSRPBinder call has already logged it
```

Comment:
that if is a bit weird - it can probably be removed 
logIssued == unsupportedSRPWarningIssued value according to line 498
and you are checking the value of unsupportedSRPWarningIssued in the other function

Reply:
This is because we have 2 paths. LogUnsupportedSRP can be called from lazily initializing the currentSRPBinder (in that case we want to log only the first time) or explicitely upon some users action (opening the graph view, explicitely trying to compile a VFX...) and in this case we want to log always.

Reply:
let us imagine unsupportedSRPWarningIssued equals to true. then 
if(logIssue || !unsupportedSRPWarningIssued ) <=> if(true || false) > then TRUE
the opposite is:
if(logIssue || !unsupportedSRPWarningIssued ) <=> if (false|| true) > then TRUE

Reply:
logIssued is just a cache to check whether the inner call to currentSRPBinder has issued the log already (logIssued is false and unsupportedSRPWarningIssued is true after the call) and in that case we don't reissue (else it will be logged twice in a row). true to false is not possible.

Reply:
shouldnt it be `forceLog` in the if statement instead of `logIssue`?

Reply:
If the test "if (logIssued || !unsupportedSRPWarningIssued)" is false, it means the message has already been log during the inner call to currentSRPBinder just above (unsupportedSRPWarningIssued was false before currentSRPBinder call, and true after), regardless of forceLog. This just prevents logging twice from the same explicit LogUnsupportedSRP call.

Reply:
okayyyyyy.... currentSRPBinder is not a getter but a function call! now all of it makes sense :)
I was going crazy
so now... how do we prevent others from going crazy? :)

Reply:
It's a getter but lazily initialized. We could move the initialization directly in the callback probably. But it was safer and less intrusive to do it this way now.